packages/components/src/button/test/index.tsx

```diff
@@ -360,33 +361,55 @@ describe( 'Button', () => {
 
 	describe( 'deprecated props', () => {
 		it( 'should not break when the legacy isPrimary prop is passed', () => {
+			// @ts-expect-error
 			render( <Button isPrimary /> );
 			expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-primary' );
 		} );
 
 		it( 'should not break when the legacy isSecondary prop is passed', () => {
+			// @ts-expect-error
 			render( <Button isSecondary /> );
 			expect( screen.getByRole( 'button' ) ).toHaveClass(
 				'is-secondary'
 			);
 		} );
 
 		it( 'should not break when the legacy isTertiary prop is passed', () => {
+			// @ts-expect-error
 			render( <Button isTertiary /> );
 			expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-tertiary' );
 		} );
 
 		it( 'should not break when the legacy isLink prop is passed', () => {
+			// @ts-expect-error
 			render( <Button isLink /> );
 			expect( screen.getByRole( 'button' ) ).toHaveClass( 'is-link' );
 		} );
 
 		it( 'should warn when the isDefault prop is passed', () => {
+			// @ts-expect-error
 			render( <Button isDefault /> );
 			expect( screen.getByRole( 'button' ) ).toHaveClass(
 				'is-secondary'
 			);
 			expect( console ).toHaveWarned();
 		} );
 	} );
+
+	describe( 'static typing', () => {
+		it( 'should have the right props', () => {
+			render( <Button href="foo" /> );
+			// @ts-expect-error - `target` requires `href`
+			render( <Button target="foo" /> );
+			// @ts-expect-error - `disabled` is only for buttons
+			render( <Button href="foo" disabled /> );
+			render( <Button href="foo" type="image/png" /> );
+			// @ts-expect-error - if button, type must be submit/reset/button
+			render( <Button type="image/png" /> );
+			// @ts-expect-error
+			render( <Button type="invalidtype" /> );
+			// @ts-expect-error - although the runtime behavior will allow this to be an anchor, this is probably a mistake.
+			render( <Button disabled __experimentalIsFocusable href="foo" /> );
```

Comment:
If it's alright, this is basically copy-pasted from your https://github.com/WordPress/gutenberg/pull/46997#pullrequestreview-1262338583.

There's no `expect()` call, which is a little strange.

But it's not really testing rendering, only typing.

Reply:
Yeah, it'd definitely be simpler to not wrap them in `render()`, and `it()`.

I can't get that to work, without getting 'JSX expressions must have one parent element':
<img width="887" alt="Screen Shot 2023-01-23 at 5 55 41 PM" src="https://user-images.githubusercontent.com/4063887/214179246-38ae822a-ce87-4279-8edd-26efade36922.png">

```jsx
	describe( 'static typing', () => {
		<Button href="foo" />

		// @ts-expect-error - `target` requires `href`
		<Button target="foo" />

		// @ts-expect-error - `disabled` is only for buttons
		<Button href="foo" disabled />

		<Button href="foo" type="image/png" />

		// @ts-expect-error - if button, type must be submit/reset/button
		<Button type="image/png" />;

		// @ts-expect-error
		<Button type="invalidtype" />

		// @ts-expect-error - although the runtime behavior will allow this to be an anchor, this is probably a mistake.
		<Button disabled __experimentalIsFocusable href="foo" />
	} );
```

The examples you [linked to](https://github.com/search?q=repo%3AWordPress%2Fgutenberg%20%22static%20typescript%22&type=code) work, though 😄

Reply:
Lol. We can use a Fragment, I guess?

```jsx
describe( 'static typing', () => {
	<>
		<Button href="foo" />
		{ /* @ts-expect-error - `target` requires `href` */ }
		<Button target="foo" />
		{ /* @ts-expect-error - `disabled` is only for buttons */ }
		<Button href="foo" disabled />
		<Button href="foo" type="image/png" />
		{ /* @ts-expect-error - if button, type must be submit/reset/button */ }
		<Button type="image/png" />
		{ /* @ts-expect-error */ }
		<Button type="invalidtype" />
		{ /* @ts-expect-error - although the runtime behavior will allow this to be an anchor, this is probably a mistake. */ }
		<Button disabled __experimentalIsFocusable href="foo" />
	</>;
} );
```

Reply:
Ah, good idea.